server/src/ankaios_server/server_state.rs

```diff
@@ -236,6 +268,15 @@ impl ServerState {
                         ));
                     }
 
+                    self.update_delete_graph(&added_workloads);
+
+                    for workload in deleted_workloads.iter_mut() {
+                        if let Some(delete_conditions) = self.delete_graph.get(&workload.name) {
+                            // [impl->swdd~server-state-adds-delete-conditions-for-deleted-workload~1]
+                            workload.dependencies = delete_conditions.clone();
+                        }
+                    }
+
```

Comment:
Maybe we can extract this to a separate method.

Background - I needed to read through the code to understand what's going on. The code is well structured and understandable, but one loses some time during maintenance. To follow the clean code principles, the code must be understandable without comments (as they get outdated) so a method with a good, descriptive name is the best option.

Reply:
Done.

Reply:
Maybe it is better to wrap the DeleteGraph into an own structure to prevent invariants and make that new structure part of ServerState. What do you think?

Reply:
I have restructured it and introduced mock calls in the ServerState tests for the DeleteGraph.